lib/http/request.ts

```diff
@@ -20,6 +20,13 @@ import { OktaAuthInterface, RequestOptions, FetchOptions, RequestData } from '..
 
 export function httpRequest(sdk: OktaAuthInterface, options: RequestOptions): Promise<any> {
   options = options || {};
+
+  if (sdk.options.httpRequestInterceptors) {
+    for (const interceptor of sdk.options.httpRequestInterceptors) {
+      interceptor(options);
```

Comment:
What is the purpose of the interceptors? To allow for manipulation of the `RequestOptions`?

Reply:
Yep, seems like there is no proper way to manipulate the url (RequestOptions) after moving the logic to sdk layer. Before ciamx fix the browser cors issue, we still need a way to proxy the request. We can keep it as an internal feature first.